lib/elixir/lib/stream.ex

```diff
@@ -455,9 +455,17 @@ defmodule Stream do
 
   """
   @spec interval(non_neg_integer) :: Enumerable.t
-  def interval(n) do
+  def interval(interval) do
+    start_time = now_in_ms()
```

Comment:
`start_time` needs to be computed when the stream starts, not when interval is called. Please see my last comment for a reference to compute it lazily on start. :)